pkg/capacity/capacity.go

```diff
@@ -59,7 +60,107 @@ func FetchAndPrint(showContainers, showPods, showUtil, showPodCount, availableFo
 	printList(&cm, showContainers, showPods, showUtil, showPodCount, showNamespace, output, sortBy, availableFormat)
 }
 
-func getPodsAndNodes(clientset kubernetes.Interface, podLabels, nodeLabels, namespaceLabels, namespace string) (*corev1.PodList, *corev1.NodeList) {
+func splitTaint(taint string) (string, string, string) {
```

Comment:
Hey @OperationalDev, thanks for this PR! I'm having a bit of trouble understanding what you're trying to do in some places, do you mind adding some more comments? For example with this function it would be very useful to have comment(s) explaining what is happening, especially what each of the output strings are intended to represent.

At a high level I thought adding support for node taints would just involve a simple loop after `nodeList` is fetched in `getPodsAndNodes` to filter out nodes that don't match the specified taints, but it looks like there's a lot more going on here. Can you help explain why we need all the additional logic here?

Reply:
Added some comments inline with the code. If there are parts that are unclear, let me know and I can add more.

Based on my initial requirement, I did just have a loop that removed nodes with taints. However on further review and testing, I realized that there are scenarios where I only want to include nodes with specific taints but remove other nodes with specific taints. Having Add/Remove Taints in one function got a bit messy which is why I split it out into multiple functions. For example I want to get capacity of nodes that have the GPU Taint but exclude nodes that have been cordoned for maintenance.